### PR TITLE
fix: incorrect volumesnapshot size

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

fix volumesnapshot restoresize bug

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
volumesnapshot always get 100Gi in RESTORESIZE before, it is because set `defaultAzureFileQuota` in using management api `CreateSnapshot` func, fix this bug using volumesource size.

test:
NAME                          READYTOUSE   SOURCEPVC       SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS       SNAPSHOTCONTENT                                    CREATIONTIME   AGE
azurefile-volume-snapshot     true         pvc-azurefile                           200Gi         csi-azurefile-vsc   snapcontent-379d8f28-6aab-4946-96d3-c88e496d95e0   2m1s           2m1s

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azurefile/controllerserver.go [hunk 1]
```diff
@@ -995,7 +995,11 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 		itemSnapshotTime = properties.LastModified()
 		itemSnapshotQuota = properties.Quota()
 	} else {
-		snapshotShare, err := d.cloud.FileClient.WithSubscriptionID(subsID).CreateFileShare(ctx, rgName, accountName, &fileclient.ShareOptions{Name: fileShareName, RequestGiB: defaultAzureFileQuota, Metadata: map[string]*string{snapshotNameKey: &snapshotName}}, snapshotsExpand)
```
Comment (andyzhangx) [inline pkg/azurefile/controllerserver.go:998]: what if just removing `RequestGiB: defaultAzureFileQuota` in L998, will that also fix the issue? that could save on Get file share call

Comment (umagnus) [reply pkg/azurefile/controllerserver.go:998]: No, it default is 0 and will return this error:
```
E1018 01:57:01.111767       1 utils.go:81] GRPC error: rpc error: code = Internal desc = create snapshot from(mc_aks-snapshot_group_aks-snapshot_eastus#f33e8e7dd87ca40fba6f47c#pvc-2e4c617d-86f9-4c7b-a8f4-097633184ca0###default) failed with storage.FileSharesClient#Create: Invalid input: autorest/validation: validation failed: parameter=fileShare.FileShareProperties.ShareQuota constraint=InclusiveMinimum value=0 details: value must be greater than or equal to 1, accountName: "f33e8e7dd87ca40fba6f47c"
```

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**